InvenTree/part/views.py

```diff
@@ -718,6 +722,151 @@ def get_initial(self):
         return initials
 
 
+class PartImport(FileManagementFormView):
+    ''' Part: Upload file, match to fields and import parts(using multi-Step form) '''
+
+    name = 'part'
+    form_steps_template = [
+        'part/import_wizard/part_upload.html',
+        'part/import_wizard/match_fields.html',
+        'part/import_wizard/match_references.html',
+    ]
+    form_steps_description = [
+        _("Upload File"),
+        _("Match Fields"),
+        _("Match References"),
+    ]
+
+    form_field_map = {
+        'name': 'name',
+        'description': 'description',
+        'keywords': 'keywords',
+        'ipn': 'ipn',
+        'revision': 'revision',
+        'link': 'link',
+        'default_expiry': 'default_expiry',
+        'minimum_stock': 'minimum_stock',
+        'units': 'units',
+        'notes': 'notes',
+        'category': 'category',
+        'default_location': 'default_location',
+        'default_supplier': 'default_supplier',
+    }
+
+    def get_field_selection(self):
+        """ Fill the form fields for step 3 """
+
+        self.file_manager.setup()
+        # collect reference indexes
+        idx_s = {}
+        for col in self.file_manager.HEADERS:
+            index = self.get_column_index(col)
+            if index >= 0:
+                idx_s[col] = index
+
+        # fetch available elements
+        self.allowed_items = {}
+        self.matches = {}
+
+        self.allowed_items['Category'] = PartCategory.objects.all()
+        self.matches['Category'] = ['name__contains']
+        self.allowed_items['default_location'] = StockLocation.objects.all()
+        self.matches['default_location'] = ['name__contains']
+        self.allowed_items['default_supplier'] = SupplierPart.objects.all()
+        self.matches['default_supplier'] = ['SKU__contains']
+
+        for row in self.rows:
+            for idx in idx_s:
+                data = row['data'][idx_s[idx]]['cell']
+
+                if idx in self.file_manager.OPTIONAL_MATCH_HEADERS:
+                    try:
+                        exact_match = self.allowed_items[idx].get(**{a:data for a in self.matches[idx]})
+                    except (ValueError, self.allowed_items[idx].model.DoesNotExist, self.allowed_items[idx].model.MultipleObjectsReturned):
+                        exact_match = None
+
+                    row['match_options_' + idx] = self.allowed_items[idx]
+                    row['match_' + idx] = exact_match
+                    continue
+
+                # general fields
+                row[idx.lower()] = data
+
+    def done(self, form_list, **kwargs):
+        """ Create items """
+        items = {}
+
+        for form_key, form_value in self.get_all_cleaned_data().items():
```

Comment:
We can maybe move this for-loop in `FileManagementFormView` as I feel this will be reused a lot

Reply:
`items` could be accessed with `self.items` instead

Reply:
will try that tomorrow.
Keeping items a funtion-scope variable is in preperation for inherited functions - keeps the effects of wrong inherited functions to this step and therefore debugable